workflows/fastqrepair.nf

```diff
@@ -21,74 +21,77 @@ workflow FASTQREPAIR {
 
     take:
     ch_samplesheet // channel: samplesheet read in from --input
-
     main:
-
     ch_versions = Channel.empty()
-    ch_multiqc_files = Channel.empty()
+    ch_decoupled = Channel.empty()
 
-    //
-    // MODULE: Run FastQC
-    //
+    // Decouple paired-end reads
+    ch_decoupled = ch_samplesheet.flatMap { metaData, filePaths -> filePaths.collect { file -> [metaData, file] } }
+
+    // Recover fastq.gz and skip *.fastq or *.fq
+    GZRT (
+        ch_decoupled
+    )
+
+    // Make fastq compliant and wipe bad characters
+    SCATTER_WIPE_GATHER (
+        GZRT.out.fastq
+    )
+
+    // Run if PAIRED-END reads only!
+    SCATTER_WIPE_GATHER.out.fixed_fastq
+        .branch {
+            single_end: it[0].single_end == true
+            paired_end: it[0].single_end == false
+        }
+        .set { filtered_ch }
+
+    // Remove unpaired reads and reads shorter than 20 nt
+    TRIMMOMATIC (
+        filtered_ch.paired_end.groupTuple()
+    )
+
+    // Settle reads pairing (re-pair)
+    BBMAPREPAIR {
+        TRIMMOMATIC.out.trimmed_reads
+    }
+
+
+    // Rename final FASTQ and REPORT files and move them into the "pickup" folder
+    RENAMER (
+        filtered_ch.single_end.concat(BBMAPREPAIR.out.interleaved_fastq),
+        SCATTER_WIPE_GATHER.out.report.groupTuple()
+    )
+
+    // Assess QC of all fastq files (both single and paired end)
     FASTQC (
```

Comment:
Have you considered running FASTQC before and after trimmomatic to make it easier to see how well it performed wehen you look at the MultiQC report?